utils/build/docker/dotnet/query-versions.fsx

```diff
@@ -35,8 +53,20 @@ module QueryVersions =
         File.WriteAllText("/app/SYSTEM_TESTS_APPSEC_EVENT_RULES_VERSION", ruleVersion)
 
     let writeWafVersion () =
-        let buffer = Native.ddwaf_get_version()
-        let version = Marshal.PtrToStringAnsi(buffer)
+        let getOldWafVersion () =
+            let mutable ver:WafVersion = WafVersion()
+            let verPtr = &&ver
+            let nativeInt = NativePtr.toNativeInt verPtr
+            NativeOld.ddwaf_get_version(nativeInt) |> ignore
+            ver.ToString()
+        
+        let getWafVersion () =
+            let buffer = Native.ddwaf_get_version()
+            Marshal.PtrToStringAnsi(buffer)
+        
+        let version = 
+            if assem.GetName().Version.Major <= 2 && assem.GetName().Version.Minor <= 14 then getOldWafVersion()
+            else getWafVersion()
```

Comment:
cant try catch here as the AccessViolationMemory exception that would result is unmanaged and wouldnt be caught. `[HandleProcessCorruptedStateExceptionsAttribute ](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.handleprocesscorruptedstateexceptionsattribute?view=net-7.0)` is deprecated